### PR TITLE
Fix/particle trajectories uniqueness

### LINKED ISSUE #1541: ValueError when doing particle trajectories
### Bug report

**Bug summary**

When trying to get particle trajectories, I get the following error:
```python
ValueError                                Traceback (most recent call last)
~/Documents/These/ramses_tests/MC_tracers/test_feedback/plot.py in <module>()
     40     dss = yt.DatasetSeries(outputs[:5], **load_kwa)
     41
---> 42     trajs = dss.particle_trajectories(indices)

~/Documents/prog/yt/yt/data_objects/time_series.py in particle_trajectories(self, indices, fields, suppress_logging)
    435         >>>     print t["particle_velocity_x"].max(), t["particle_velocity_x"].min()
    436         """
--> 437         return ParticleTrajectories(self, indices, fields=fields, suppress_logging=suppress_logging)
    438
    439 class TimeSeriesQuantitiesContainer(object):

~/Documents/prog/yt/yt/data_objects/particle_trajectories.py in __init__(self, outputs, indices, fields, suppress_logging)
    121         for field in ("particle_position_%s" % ax for ax in "xyz"):
    122             for i, (fn, (time, indices, pfields)) in enumerate(sorted(my_storage.items())):
--> 123                 output_field[indices, i] = pfields[field]
    124             self.field_data[field] = array_like_field(
    125                 dd_first, output_field.copy(), fds[field])

ValueError: shape mismatch: value array of shape (10096,) could not be broadcast to indexing result of shape (96,)
```

**Code for reproduction**

```python
import yt
from glob import glob

def load_filters(ds):
    ptypes = {
        0: 'DM',
        1: 'star',
        100: 'MC_tracer',
        101: 'star_tracer',
        123: 'old_star_tracer'
    }

    def gen_filter(pt):
        def loc(pfilter, data):
            return data[(pfilter.filtered_type, "particle_type")] == pt
        return loc

    for pt in ptypes:
        yt.add_particle_filter(ptypes[pt], function=gen_filter(pt),
                               requires=['particle_type'], filtered_type='all')

    for pt in ptypes:
        ds.add_particle_filter(ptypes[pt])


if __name__ == '__main__':
    load_kwa = dict(extra_particle_fields=[('particle_type', 'I')])
    outputs = sorted(glob('output_*/info*.txt'))

    # Get the index of the particles released by stars
    last_ds = yt.load(outputs[-1], **load_kwa)
    load_filters(last_ds)

    indices = last_ds.r['old_star_tracer', 'particle_identifier'].astype(int)

    # Now let's take a look at their life
    dss = yt.DatasetSeries(outputs, **load_kwa)

    trajs = dss.particle_trajectories(indices)
```

**Version Information**
  * Linux x64
  * Python Version: 3.6
  * yt version: 3.5dev0

I have installed yt from source. The dataset is attached.
[dataset.zip](https://github.com/yt-project/yt/files/1228827/dataset.zip)

### PR DESCRIPTION
**Pull request by cphyc**
## PR Summary
This fixes #1541.

## PR Checklist

- [x] Code passes flake8 checker
- [x] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

File: yt/data_objects/particle_trajectories.py [hunk 1]
```diff
@@ -120,7 +121,10 @@ def __init__(self, outputs, indices, fields=None, suppress_logging=False):
         output_field.fill(np.nan)
         for field in ("particle_position_%s" % ax for ax in "xyz"):
             for i, (fn, (time, indices, pfields)) in enumerate(sorted(my_storage.items())):
-                output_field[indices, i] = pfields[field]
+                try:
+                    output_field[indices, i] = pfields[field]
+                except ValueError:
+                    raise YTIllDefinedParticleData('The dataset contains duplicate particle index')
```
Comment (jzuhone) [inline yt/data_objects/particle_trajectories.py]: I would change the error message to say "This dataset contains duplicate particle indices!"

File: yt/data_objects/tests/test_particle_trajectories.py [hunk 2]
```diff
@@ -43,3 +47,22 @@ def test_etc_traj():
...
+
+    ts = DatasetSeries([fake_particle_ds(fields=fields, negative=negative, units=units,
+                                         npart=n_particles, data=data)
+                        for i in range(n_steps)])
+    try:
+        traj = ts.particle_trajectories([0])
+    except YTIllDefinedParticleData:
+        pass
+    else:
+        assert False
```
Comment (ngoldbaum) [inline yt/data_objects/tests/test_particle_trajectories.py]: it might be cleaner to use `assert_raises` here

Comment (cphyc) [reply yt/data_objects/tests/test_particle_trajectories.py]: Didn't know!

**Comment by cphyc** [comment]
Any guess why the py3 tests failed? This seem to be unrelated to the PR.

**Comment by ngoldbaum** [comment]
@yt-fido test this please

**Comment by ngoldbaum** [comment]
See #1702.

**Pull request merged**